File: SOURCE_EXTERNAL/VecompSoftware.DocSuite.ExternalViewer/src/app/mappers/commons/category.mapper.ts
import { Injectable } from '@angular/core'; 

import { BaseMapper } from '../base.mapper'; 
import { CategoryModel } from '../../models/commons/category.model'; 

@Injectable()
export class CategoryMapper implements BaseMapper {

    mapFromJson(json: any): CategoryModel {

        if (!json) {
            return null;
        }

        let model: CategoryModel = new CategoryModel();

        model.name = json.Name;
        model.active = json.isActive;
        model.code = json.Code;
        model.hierarchyCode = json.HierarchyCode;
        model.hierarchyDescription = json.HierarchyDescription;

        return model;
    }

}

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Mapper.Entity/Commons/Implementation/CategoryMapper.cs
using VecompSoftware.DocSuiteWeb.Entity.Commons;

namespace VecompSoftware.DocSuiteWeb.Mapper.Entity.Commons
{
    public class CategoryMapper : BaseEntityMapper<Category, Category>, ICategoryMapper
    {
        public CategoryMapper()
        {

        }

        public override Category Map(Category entity, Category entityTransformed)
        {
            #region [ Base ]
            entityTransformed.Name = entity.Name;
            entityTransformed.IsActive = entity.IsActive;
            entityTransformed.Code = entity.Code;
            entityTransformed.FullCode = entity.FullCode;
            entityTransformed.FullIncrementalPath = entity.FullIncrementalPath;
            entityTransformed.StartDate = entity.StartDate;
            entityTransformed.EndDate = entity.EndDate;
            #endregion

            return entityTransformed;
        }

    }
}
